### ISSUE TITLE
Exception '*** -[__NSArrayM objectAtIndex:]

### ISSUE THREAD

**Issue by selimonline** [MostWantIT/react-native-video-editor#6]
when I am calling  RNVideoEditor.merge with necessary params. It gives following erros...

Exception '*** -[__NSArrayM objectAtIndex:]: index 0 beyond bounds for empty array' was thrown while invoking merge on target RNVideoEditor with params (
        (
        "file:///private/var/mobile/Containers/Data/Application/0AC17BF8-977F-42B9-8444-55E313599228/tmp/YFHn4jXxSe7C-SCVideo.0.mp4",
        "file:///private/var/mobile/Containers/Data/Application/0AC17BF8-977F-42B9-8444-55E313599228/tmp/YFHn4jXxSe7C-SCVideo.1.mp4"
    ),
    1582,
    1583
)

**Comment by Antonides** [comment]
What library did you use to get the mp4 files? Did the example work for you?

**Comment by selimonline** [comment]
https://github.com/maxs15/react-native-screcorder
this library I am using to capture video....

I checkout node_modules/react-native-video-editor/ios/RNVideoEditor.m files
where I get 
for (id object in fileNames)
    {
        
        AVAsset *asset = [AVAsset assetWithURL:[NSURL fileURLWithPath:object]];

this asset always giving null return... that's 
[videoTrack insertTimeRange:timeRange
                            ofTrack:[[asset tracksWithMediaType:AVMediaTypeVideo] objectAtIndex:0]
                             atTime:insertTime
                              error:nil];
***objectAtIndex:0 returning error
It seems, file path (file:///private/var/mobile/Containers/Data/Application/0AC17BF8-977F-42B9-8444-55E313599228/tmp/YFHn4jXxSe7C-SCVideo.0.mp4) not workin here 
 AVAsset *asset = [AVAsset assetWithURL:[NSURL fileURLWithPath:object]];

but why..? any idea?

**Comment by Antonides** [comment]
Can you try remove the file:// from the url string.

**Comment by selimonline** [comment]
yes I tried that way already... found somewhere in google... but did not work...
like 
/private/var/mobile/Containers/Data/Application/0AC17BF8-977F-42B9-8444-55E313599228/tmp/YFHn4jXxSe7C-SCVideo.0.mp4)

**Comment by Antonides** [comment]
can you try to switch to .mov file? 

I try the example. And it works. only difference with you is that i have .mov file.

/private/var/mobile/Containers/Data/Application/76616072-DB7D-4DB4-AB9D-3503E717F5A2/tmp/2E262EE9-A8F3-4D9D-8A6D-BA295914B48E-337-00000011968BB88C.mov

**Comment by selimonline** [comment]
I did not try mov... as all my steps of my app using mp4 with so many libraries... but if mov works, mp4 should work too....

**Comment by Antonides** [comment]
Can you try if the example works for you?

**Comment by selimonline** [comment]
yes I tried example demo, where you use 'react-native-camera' library.. 
and that  creates .mov files tooo...
but unfortunately that gives exactly same error...
look at my screenshot 
https://image.prntscr.com/image/b29P2I23RsCQlKS6gE-KoQ.png

 AVAsset *asset = [AVAsset assetWithURL:[NSURL fileURLWithPath:object]];

this gives null return as like before... 
Can you please suggest me any other options. I think, it's a bug. I don't know objective c. that's why I can't make any change in  AVAsset *asset
BTW, RN Version .48

and thanks.. I really appreciate your valuable time...

**Comment by andrewscofield** [comment]
Hey, I also had this bug and found a fix for it. In my case I was trying to merge video files with no audio on them. The solution that worked for me was to comment out references to the audio track:

https://github.com/MostWantIT/react-native-video-editor/blob/master/ios/RNVideoEditor.m#L42-L43

and here

https://github.com/MostWantIT/react-native-video-editor/blob/master/ios/RNVideoEditor.m#L59-L62


@Antonides - Maybe there is a way to detect if the video being merged has audio? Or perhaps a new parameter to explicitly set an audio track or not? I don't know what the best way is.

**Comment by selimonline** [comment]
dear, in my case. I have both audio/video. Still  getting that error... Exception '*** -[__NSArrayM objectAtIndex:]

**Comment by andrewscofield** [comment]
Ah, sorry man. I was having the exact same error and this fixed it. Are you 100% sure you have audio?

Also, there is only one other place there is a objectAtIndex: https://github.com/MostWantIT/react-native-video-editor/blob/master/ios/RNVideoEditor.m#L107

And it has to do with the paths of the files, so maybe something there? Wish I couple help more!

**Comment by selimonline** [comment]
yes, I am 100% sure, I have audio. I don't know, almost anything of C, But as per my debug, problem is here 
for (id object in fileNames)
    {
        
        AVAsset *asset = [AVAsset assetWithURL:[NSURL fileURLWithPath:object]];
        
        CMTimeRange timeRange = CMTimeRangeMake(kCMTimeZero, asset.duration);
        
        [videoTrack insertTimeRange:timeRange
                            ofTrack:[[asset tracksWithMediaType:AVMediaTypeVideo] objectAtIndex:0]
                             atTime:insertTime
                              error:nil];
        
        [audioTrack insertTimeRange:timeRange
                            ofTrack:[[asset tracksWithMediaType:AVMediaTypeAudio] objectAtIndex:0]
                             atTime:insertTime
                              error:nil];
        
        insertTime = CMTimeAdd(insertTime,asset.duration);
}

I tried to work removing audio part here. still video part gives that error...
Any way, thanks for your helping hands...

**Comment by bsiddiqui** [comment]
@salimbutex did you figure this out?

**Comment by juandvallero** [comment]
I had fix this issue by replacing in the video path:
```
if (this.clips[0] && this.clips[1]) {
      let v1 = this.clips[0].replace("%2540", "%40").replace("%252F", "%2F");
      let v2 = this.clips[1].replace("%2540", "%40").replace("%252F", "%2F");

      RNVideoEditor.merge(
        [v1, v2],
        results => {
          alert("Error: " + results);
        },
        (results, file) => {
          alert("Success : " + results + " file: " + file);
        }
      );
    }
  }
```

**Comment by BenBrostoff** [comment]
This issue can also happen when the `AVAsset` has a `duration` of zero. I'm still working through this issue, but I was finding with Expo that `Expo.ImagePicker.launchImageLibraryAsync` will at first return file URLs that are recognized by `AVAsset` but given zero duration, and then on subsequent runs have positive duration and work fine.

**Comment by BenBrostoff** [comment]
Because this issue (at least for me) is intermittent and the result changes *given the same URIs*, my best guess right now is that it has to do with [how `AVAsset` works](https://developer.apple.com/library/archive/documentation/AudioVideo/Conceptual/AVFoundationPG/Articles/01_UsingAssets.html) and loads video properties. If duration is only sometimes available, the solution would seem to be to force it to be there when calling `asset tracksWithMediaType:AVMediaTypeAudio`. Just a guess at this point; I can open a PR if it works.

**Comment by BenBrostoff** [comment]
Sorry for posting three times on this issue - my problem was file permissions related and simply removing `file://` resolved my issue:

```
NSString *newFilepath = [filepath stringByReplacingOccurrencesOfString:@"file://" withString:@""];
```

**Comment by BhuwaneshSisodia** [comment]
By placing **.replace("file://", "").replace("file://", "");**** it will solve the issue. 
```
mergeVideos() {
     let v1 = this.state.video[0].uri.replace("file://", "").replace("file://", "");
     let v2 = this.state.video[1].uri.replace("file://", "").replace("file://", "");
RNVideoEditor.merge(
     [v1,v2],
     results => {
          alert("Error: " + results);
     },
     (results, file) => {
          alert("Success : " + results + " file: " + file);
          }
     );
}

```

**Comment by jrartiga** [comment]
Hey Guys,
do you have any idea why I'm having this issue,
basically what I have are two videos, one was recorded with the back camera, and the other was recorded with the front camera, and when I tried merged them  this error comes up.

![image](https://user-images.githubusercontent.com/6892349/71262659-ddba3c80-2305-11ea-857c-88b61c281ed0.png)

And it's kinda weird since, I have merged two videos recorded with front camera or both with back camera, everything is good.
